Gulpfile.js

```diff
@@ -0,0 +1,455 @@
+'use strict';
+/*jslint node: true */
+/*jslint esversion: 6*/
+//TODO(shuwen): split into multiple files
+//TODO(shuwen): replace path string concatenation with path.join
+
+var fs = require('fs');
+var gulp = require('gulp');
+var glob = require('glob');
+var gutil = require('gulp-util');
+var del = require('del');
+var sass = require('gulp-sass');
+var rename = require('gulp-rename');
+var merge = require('merge-stream');
+var path = require('path');
+var vulcanize = require('gulp-vulcanize');
+var debug = require('gulp-debug');
+var run = require('run-sequence');
+var htmlmin = require('gulp-minify-html');
+var inlinemin = require('gulp-minify-inline');
+var wrap = require('gulp-wrap');
+var inlineAssets = require('gulp-inline-assets');
+var marked = require('gulp-marked');
+var changed = require('gulp-changed');
+var es = require('event-stream');
+var cache = require('gulp-cached');
+var postcss = require('gulp-postcss');
+var autoprefixer = require('autoprefixer');
+var git = require('gulp-git');
+var bump = require('gulp-bump');
+var tagVersion = require('gulp-tag-version');
+var conventionalChangelog = require('gulp-conventional-changelog');
+var ghPages = require('gulp-gh-pages');
+var through = require('through2');
+var hogan = require('hogan.js');
+var header = require('gulp-header');
+var base64 = require('gulp-base64');
+var minimist = require('minimist');
+
+var SRC = 'src/';
+var BUILD = 'build/';
+var BUILD_DOCS = 'build_docs/';
+var DIST = 'dist/';
+var TEMPLATES = 'gulp/templates/'; //TODO swap to gulp
+
+var PATCH_LIST = [
+	'bower_components/moment/min/moment.min.js'
+];
+
+/** BUILD **/
+
+gulp.task('patch-lib', function() {
+	gulp.src(PATCH_LIST, {base: path.join(__dirname, 'bower_components')})
+		.pipe(debug())
+		.pipe(wrap("(function(define, require) { {{{contents}}} })();",{},{engine:"hogan"}).on('error',console.error))
+		.pipe(gulp.dest( path.join(__dirname, 'bower_components') ));
+});
+
+gulp.task('clean', function() {
+	return del([BUILD + '**', BUILD_DOCS+ '**', DIST+ '**']);
+});
+
+gulp.task('copy', function() {
+	return gulp.src([SRC + '**/*.+(html|js|woff)', '!' + SRC +'**/example.html'])
+		.pipe(changed(BUILD))
+		.pipe(debug())
+		.pipe(gulp.dest(BUILD));
+});
+
+gulp.task('sass', function() {
+	var wrapper = fs.readFileSync(TEMPLATES + "style_module_template.html");
```

Comment:
@shuwen looks like some strange mangling happening with the style module `<!--.mm-icon *.mm-icon @license.mm-icon *.mm-icon Copyright.mm-icon (c).mm-icon 2015.mm-icon MediaMath.mm-icon Inc..mm-icon All.mm-icon rights.mm-icon reserved..mm-icon *.mm-icon This.mm-icon code.mm-icon may.mm-icon only.mm-icon be.mm-icon used.mm-icon under.mm-icon the.mm-icon BSD.mm-icon style.mm-icon license.mm-icon found.mm-icon at.mm-icon http.mm-icon://mediamath.github.io/strand/LICENSE.txt --.mm-icon> <dom-module.mm-icon id="-style".mm-icon> <template.mm-icon> <style.mm-icon>` etc. Noticed this in docs.